typescript/collections/DefaultArray/DefaultArray.ts

```diff
@@ -0,0 +1,4 @@
+// returns an Array of `length` elements filled with `value`
+// `value' defaults to `0`
+export const defaultArray = (length: number, value: any = 0) =>
+  Array.from({ length }, () => value)
```

Comment:
Hey hey, again thanks for your contribution, and I apologize for slow replies.

I don't know about this function, really. To me it looks like it's roughly the same amount of syntax as writing `Array.from(...)`, and it's less flexible so e.g. filling an array with random values will be substantially different than filling it with zeroes.

Reply:
Well, like I said - it's your project, you know better what you want it to be, just saw your comment somewhere in Neatness :)

I can still extend this to accept functions though

Reply:
Ah, the comment was about [`defaultdict`][1] in Python, which has unique semantics: it's a mapping with autovivification, kind of like objects in Perl, where you can access a field and if it's not there, a default value is constructed and returned.

It's possible to implement in JS, using a proxy object for example.

[1]: https://docs.python.org/3/library/collections.html#collections.defaultdict